Loop/Managers/AlertPermissionsChecker.swift

```diff
@@ -0,0 +1,108 @@
+                                                             backgroundContent: criticalAlertPermissionsAlertContent,
+                                                             trigger: .immediate)
+    
+    private weak var alertManager: AlertManager?
+    
+    private var isAppInBackground: Bool {
+        return UIApplication.shared.applicationState == UIApplication.State.background
+    }
+    
+    lazy private var cancellables = Set<AnyCancellable>()
+
+    init(alertManager: AlertManager) {
+        self.alertManager = alertManager
+        
+        // Check on loop complete, but only while in the background.
+        NotificationCenter.default.publisher(for: .LoopCompleted)
+            .receive(on: RunLoop.main)
+            .sink { [weak self] _ in
+                guard let self = self else { return }
+                if self.isAppInBackground {
+                    self.check()
+                }
+            }
+            .store(in: &cancellables)
+        
+        // Check on app resume
+        NotificationCenter.default.publisher(for: UIApplication.willEnterForegroundNotification)
+            .receive(on: RunLoop.main)
+            .sink { [weak self] _ in
+                self?.check()
+            }
+            .store(in: &cancellables)
+
+        NotificationCenter.default.publisher(for: UIApplication.didEnterBackgroundNotification)
+            .receive(on: RunLoop.main)
+            .sink { [weak self] _ in
+                self?.check()
+            }
+            .store(in: &cancellables)
+    }
+
+    func check() {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            DispatchQueue.main.async {
+                let notificationsPermissions = settings.alertSetting
+                let criticalAlertsPermissions = settings.criticalAlertSetting
+                
+                if notificationsPermissions == .disabled || criticalAlertsPermissions == .disabled {
+                    self.maybeNotifyPermissionsDisabled()
+                } else {
+                    self.permissionsEnabled()
+                }
+            }
+        }
+    }
+    
+    private func maybeNotifyPermissionsDisabled() {
+        if !UserDefaults.standard.hasUserBeenNotifiedOfPermissionsAlert {
+            alertManager?.issueAlert(AlertPermissionsChecker.criticalAlertPermissionsAlert)
+            UserDefaults.standard.hasUserBeenNotifiedOfPermissionsAlert = true
+        }
+    }
+    
+    private func permissionsEnabled() {
+        alertManager?.retractAlert(identifier: AlertPermissionsChecker.criticalAlertPermissionsAlertIdentifier)
+        UserDefaults.standard.hasUserBeenNotifiedOfPermissionsAlert = false
+    }
+    
+}
+
+extension UserDefaults {
+    
+    private enum Key: String {
+        case hasUserBeenNotifiedOfPermissionsAlert = "com.loopkit.Loop.AlertPermissionsChecker"
```

Comment:
[nit] Consider "com.loopkit.Loop.HasUserBeenNotifiedOfPermissionsAlert". "AlertPermissionsChecker" really doesn't indicate what it is for.